### PR TITLE
KB Grid handle empty popup footer

### PR DESCRIPTION
**Pull request by ntacheva**

### CODE REVIEW

File: knowledge-base/grid-handle-empty-popup-footer.md [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+When I am using the [Popup Form Template]({%slug grid-templates-popup-form%}) in the Grid for Blazor, the [default Update and Cancel buttons are removed]({%slug grid-templates-popup-form%}#specifics). I am not using the [`ButtonsTemplate`]({%slug grid-templates-popup-buttons%}) and the footer remains empty. This takes unnecessary space and affects the UI's aesthetics.
```
Comment (yordan-mitev) [inline knowledge-base/grid-handle-empty-popup-footer.md:25]: Use **bold** when mentioning the labels of UI components.
```suggestion
When I am using the [Popup Form Template]({%slug grid-templates-popup-form%}) in the Grid for Blazor, the [default **Update** and **Cancel** buttons are removed]({%slug grid-templates-popup-form%}#specifics). I am not using the [`ButtonsTemplate`]({%slug grid-templates-popup-buttons%}) and the footer remains empty. This takes unnecessary space and affects the UI's aesthetics. 
```

File: knowledge-base/grid-handle-empty-popup-footer.md [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+When I am using the [Popup Form Template]({%slug grid-templates-popup-form%}) in the Grid for Blazor, the [default Update and Cancel buttons are removed]({%slug grid-templates-popup-form%}#specifics). I am not using the [`ButtonsTemplate`]({%slug grid-templates-popup-buttons%}) and the footer remains empty. This takes unnecessary space and affects the UI's aesthetics. 
+
+The requirement is to either hide this empty footer or utilize it by placing the custom form buttons within.
```
Comment (yordan-mitev) [inline knowledge-base/grid-handle-empty-popup-footer.md:27]:
```suggestion
I want to either hide this empty footer or utilize it by placing the custom form buttons within. 
```

File: knowledge-base/grid-handle-empty-popup-footer.md [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+
+This KB article also answers the following questions:
+
+- How to hide the empty footer in the edit popup of the Grid for Blazor?
+- How to display custom buttons in the footer of the edit popup?
+- How to manage form submission with custom buttons in the Grid for Blazor?
+
+## Solution
+
+There are two approaches to address this requirement:
```
Comment (yordan-mitev) [inline knowledge-base/grid-handle-empty-popup-footer.md:37]:
```suggestion
To hide the empty footer or utilize it by placing the custom form buttons, choose either of the following approaches:
```

File: knowledge-base/grid-handle-empty-popup-footer.md [hunk 4]
```diff
@@ -0,0 +1,460 @@
+---
+title:  Hiding or Removing the Empty Popup Footer when Using the Popup Form Template
```
Comment (dimodi) [inline knowledge-base/grid-handle-empty-popup-footer.md:2]: I would shorten this. A / The are not required in headings, and you can add more keywords elsewhere.

```suggestion
title:  Remove Empty Popup Footer when Using Popup Form Template
```

File: knowledge-base/grid-handle-empty-popup-footer.md [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+
+    protected override void OnInitialized()
+    {
+        LoadData();
+    }
+
+    private void LoadData()
+    {
+        GridData = GetPeople();
+    }
```
Comment (dimodi) [inline knowledge-base/grid-handle-empty-popup-footer.md:154]: This looks over-engineered. Why not generate the data in OnInitialized and forget about it?

**Review by dimodi — APPROVED** [review]

**Pull request merged**